### PR TITLE
Fix Health Checks section in installation readme

### PR DESCRIPTION
**Pull request by andythsu**
## Description

Fix the wording in Health Checks section to avoid confusion

## Additional context and related issues

https://trinodb.slack.com/archives/C059KUNPTSP/p1733245765766909

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -318,9 +318,10 @@ reference documentation](https://github.com/trinodb/charts/blob/main/charts/gate
 
 ### Health Checks
 
-Trino Gateway checks the health of each backend and **deactivates it if 
-unhealthy**. A backend that fails a health check must be manually reset to 
-active. Automatic recovery is not supported.
+Trino Gateway checks the health of each backend and maintains an in-memory state for each backend.
+A backend that fails the health check will be set to UNHEALTHY and trino gateway will not route to
+such backends. This is independent of the active/inactive status.
```
Comment (Chaho12) [inline docs/installation.md:323]: do we have docs on what active/inactive status mean? if we do, how about adding a hyperlink to this.

**Review by Chaho12 — APPROVED** [review]
LGTM 👍

File: docs/installation.md (hunk 1 above)
Comment (andythsu) [reply docs/installation.md:323]: no.. i don't think so. We do have other references on `active`, but we never really have a proper definition for it

**Review by mosabua — CHANGES_REQUESTED** [review]
Can you fix

* wrapping at 80 chars column
* avoid future tense (no "will")
* dont use "trino gateway" .. lower case
* Dont use "Please"
* Link just with `(routing-rules..` no `./`
* Maybe document the statuses better if needed

**Comment by andythsu** [comment]
> Can you fix
> 
> * wrapping at 80 chars column
> * avoid future tense (no "will")
> * dont use "trino gateway" .. lower case
> * Dont use "Please"
> * Link just with `(routing-rules..` no `./`
> * Maybe document the statuses better if needed

fixed.

**Review by mosabua — APPROVED** [review]
Looks good now. Thank you @andythsu !

**Pull request merged**